src/lib/viewers/media/MediaBaseViewer.js

```diff
@@ -387,13 +414,57 @@ class MediaBaseViewer extends BaseViewer {
      */
     togglePlay() {
         if (this.mediaEl.paused) {
-            this.mediaEl.play();
-            this.emit('play');
-            this.handleRate();
-            this.handleVolume();
+            this.play();
         } else {
-            this.mediaEl.pause();
-            this.emit('pause');
+            this.pause();
+        }
+    }
+
+    /**
+     * Validates time parameter
+     *
+     * @private
+     * @param {number} time - time for media
+     * @return {boolean} - true if time is valid
+     */
+    isValidTime(time) {
+        return !isNaN(time) && isFinite(time) && time >= 0 && time <= this.mediaEl.duration;
+    }
+
+    /**
+     * Pauses segment at certain time
+     *
+     * @private
+     * @param {number} time - time for media
+     * @emits pause
+     * @return {void}
+     */
+    pauseSegmentAt(time) {
+        // Remove earlier event listener
+        this.mediaEl.removeEventListener('timeupdate', this.pauseSegmentListener);
```

Comment:
Could this be removed inside the pauseSegmentListener callback? Then we don't have to remove it again in pause()

Reply:
We do need `removeEventListener()` again in pause because user may have manually pause the video in the middle of the segment. We don't want to pause after user resumes.